Userland/Libraries/LibAudio/FlacLoader.cpp

```diff
@@ -0,0 +1,899 @@
+{
+    m_stream->seek(position);
+}
+
+// TODO implement these
+RefPtr<Buffer> FlacLoaderPlugin::get_more_samples([[maybe_unused]] size_t max_bytes_to_read_from_input)
+{
+    Vector<Frame> samples;
+    size_t remaining_samples = max_bytes_to_read_from_input;
+    while (remaining_samples > 0) {
+        if (!m_current_frame.has_value()) {
+            next_frame();
+            if (!m_error_string.is_empty()) {
+                dbgln("Frame parsing error: {}", m_error_string);
+                return nullptr;
+            }
+            // HACK: Test the start of the next subframe
+            // auto input = m_stream->bit_stream();
+            // u64 next = input.read_bits_big_endian(64);
+            // dbgln("After frame end: {}", next);
+        }
+        samples.append(m_current_frame_data.take_first());
+        if (m_current_frame_data.size() == 0) {
+            m_current_frame.clear();
+        }
+        --remaining_samples;
+    }
+
+    return Buffer::create_with_samples(move(samples));
+}
+
+void FlacLoaderPlugin::next_frame()
+{
+    bool ok = true;
+    InputBitStream bit_stream = m_stream->bit_stream();
+#define CHECK_OK(msg)                                                                                                      \
+    do {                                                                                                                   \
+        if (!ok) {                                                                                                         \
+            m_error_string = String::formatted("Frame parsing failed: {}", msg);                                           \
+            bit_stream.align_to_byte_boundary();                                                                           \
+            dbgln_if(AFLACLOADER_DEBUG, "Crash in FLAC loader: next bytes are {:x}", bit_stream.read_bits_big_endian(32)); \
+            return;                                                                                                        \
+        }                                                                                                                  \
+    } while (0)
+
+#define CHECK_ERROR_STRING                                             \
+    do {                                                               \
+        if (!m_error_string.is_null() && !m_error_string.is_empty()) { \
+            ok = false;                                                \
+            CHECK_OK(m_error_string);                                  \
+        }                                                              \
+    } while (0)
+
+    // Crypto::Checksum::CRC8_CCITT computed_checksum;
+    // auto checksum_update = [&](u16 value) {
+    //     ByteBuffer bytes = ByteBuffer::create_uninitialized(2);
+    //     bytes.bytes()[0] = (value >> 8) & 0xff;
+    //     bytes.bytes()[1] = (value >> 0) & 0xff;
+    //     computed_checksum.update(bytes);
+    // };
```

Comment:
Please avoid committing commented-out code.
if you don't want to implement this atm, just leave a FIXME.